CMakeLists.txt

```diff
@@ -24,6 +24,11 @@ if(BUILD_DOCS OR BUILD_DOCSONLY)
     add_subdirectory(docs)
 endif()
 
+find_library(CRYPTO_LIB crypto)
+if(NOT CRYPTO_LIB)
+  message(FATAL_ERROR "crypto library not found")
+endif()
+
```

Comment:
- This should be optional, and the build should use the bundled implementation when libcrypto was not found.
- It should use the cmake module [FindOpenSSL](https://cmake.org/cmake/help/v3.21/module/FindOpenSSL.html) and lbhashkit should link against `OpenSSL::Crypto` if found.
- It should be located in `src/libhashkit/CMakeLists.txt`